src/sentry/api/endpoints/relay/project_configs.py

```diff
@@ -59,6 +69,42 @@ def _post(self, request: Request):
         else:
             return Response("Unsupported version, we only support versions 1 to 3.", 400)
 
+    def _post_or_schedule_by_key(self, request: Request):
+        public_keys = set(request.relay_request_data.get("publicKeys") or ())
+
+        configs = {}
+        pending = []
+        for key in public_keys:
+            computed = self._get_cached_or_schedule(key)
+            if not computed:
+                pending.append(key)
+            else:
+                configs[key] = computed
+
+        if len(pending) > 0:
+            configs["pending"] = pending
+
+        return Response(configs, status=200)
+
+    def _get_cached_or_schedule(self, public_key) -> Optional[dict]:
+        """
+        Returns the config of a project if it's in the cache; else, schedules a
+        task to compute and write it into the cache.
+
+        Debouncing of the project happens after the task has been scheduled.
+        """
+        cached_config = projectconfig_cache.get(public_key)
```

Comment:
Thinking out loud: We could run minimal validation here to ensure that the public key at least _looks_ like a valid key, i.e. it has the right length and character set. It's very unlikely that someone sends invalid values in here, however, since Relay already performs validation and the endpoint isn't used by end users. So I think it's fine to go without.

Reply:
What consequences has the usage of invalid keys, besides unnecessary reads from redis?